src/components/tracked-controls.js

```diff
@@ -35,6 +36,27 @@ module.exports.Component = registerComponent('tracked-controls', {
 
     this.previousControllerPosition = new THREE.Vector3();
     this.updateGamepad();
+    this.onComponentInitialized = bind(this.onComponentInitialized, this);
+  },
+
+  play: function () {
+    this.el.addEventListener('componentinitialized', this.onComponentInitialized);
+  },
+
+  pause: function () {
+    this.el.removeEventListener('componentinitialized', this.onComponentInitialized);
+  },
+
+  onComponentInitialized: function (evt) {
+    if (evt.detail.name !== 'tracked-controls') { return; }
+    var self = this;
+    Object.keys(this.el.components).forEach(function (name) {
+      const component = self.el.components[name];
+      if (component.controllerPresent) {
+        // Emit controllerpresent event.
+        self.el.emit('controllerpresent', {name: name, component: component});
```

Comment:
So the general problem is this:

Assume some tracked controllers would be available through Gamepad API.
But they are not available until actually used, due to security restrictions.

Current `oculus-touch-controls` and `vive-controls` usage is top-down declarative (declare the components to assert they are allowed to be present, and components check whether really present) so `tracked-controls` is only injected upon detection by appropriate component; unless they want to be constantly checking controller list for new entries all the time, they would need to listen to new-controller event which is `gamepadconnected`.

During original discussions, the core team preferred top-down declarative to bottom-up (e.g. `auto-detect-controllers` at the time due to minimal point-in-time concerns over scalability of pattern, and overall fit with other A-Frame component usage.

If we want to preserve top-down declarative, but attempt to abstract away completely from Gamepad API with `tracked-controls`, then the `tracked-controls` system needs to be active regardless of whether any component instances exist, and needs to wrap or transform any data the Gamepad API would provide, including connected/disconnected events.